controllers/machinehealthcheck_controller.go

```diff
@@ -333,6 +333,10 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 func (r *MachineHealthCheckReconciler) patchHealthyTargets(ctx context.Context, logger logr.Logger, healthy []healthCheckTarget, m *clusterv1.MachineHealthCheck) []error {
 	errList := []error{}
 	for _, t := range healthy {
+		// machine always in healthy status
+		if conditions.IsTrue(t.Machine, clusterv1.MachineHealthCheckSuccededCondition) {
+			continue
+		}
```

Comment:
🤔 Would this check make the whole reconciliation non re-entrant? What happens if there is a remediation request but the machine was somehow marked as MachineHealthCheckSuccededCondition?

Reply:
Right, the way this is in the PR if e.g [deleting the remediation request fails](https://github.com/kubernetes-sigs/cluster-api/pull/5729/files#diff-34cc13ef9d81a69fe97139ffce42c13fbf1ce1d330b00db41f9b29fcde9baaa4R353)
we set `MachineHealthCheckSuccededCondition` as true anyway and the next iteraction we don't retry to delete